src/Cache.cc

```diff
@@ -584,12 +588,15 @@ Cache::getRoomInfo(const std::vector<std::string> &rooms)
 
         for (const auto &room : rooms) {
                 lmdb::val data;
+                auto statesdb  = getStatesDb(txn, room);
 
                 // Check if the room is joined.
                 if (lmdb::dbi_get(txn, roomsDb_, lmdb::val(room), data)) {
                         try {
                                 RoomInfo tmp = json::parse(std::string(data.data(), data.size()));
                                 tmp.member_count = getMembersDb(txn, room).size(txn);
+                                tmp.join_rule = getRoomJoinRules(txn, statesdb);
```

Comment:
Could you copy those lines in `Cache::singleRoomInfo` also, and make the `RoomSettings` page call that method instead of `Cache::getRoomInfo`.